File: 1338. Reduce Array Size to The Half/_1338.py
class Solution:
    def minSetSize(self, arr) -> int:
        myDict = dict()
        half = len(arr)/2

        for i in arr:
            tmp = myDict.get(i, 0)
            myDict[i] = tmp+1

        value = 0
        count = 0
        for k in sorted(myDict,key=myDict.__getitem__,reverse=True):
            value += myDict[k]
            count += 1
            if value>=half:
                return count

File: 1338. Reduce Array Size to The Half/_1338.go
package leetcode

func minSetSize(arr []int) int {
    halfLen := (len(arr)+1)/2
	myDict := make(map[int]int, 0)
	for _,v := range arr{
		myDict[v]++
	}

	count := 0
	value := 0

	res := make([]int, 0, len(myDict))
	for _,v := range myDict{
		res = append(res, v)
	}

	sort.Ints(res)

	for i:=len(res)-1; i>=0; i--{
		value += res[i]
		count++
		if value >= halfLen{
			return count
		}
	}
	return count
}